### ISSUE TITLE
Further work needed to deprecate `"melee_cut"`

### ISSUE THREAD

**Issue by Maleclypse** [CleverRaven/Cataclysm-DDA#57939]
### Is your feature request related to a problem? Please describe.

During #57913 where I removed most instances of `"melee_cut"` in favor of the more flexible `"melee_damage"' array it was determined that "melee_damage" is not used in assign and therefore cannot be included in `"relative"` and `"proportional"` json arrays.

### Solution you would like.

For "melee_damage" to be usable in proportional and relative monster copies.   I'd be fine with it either being a difference applied across the existing damage types in the copied array or it being specifiable by damage type so that if it had multiple damage types you could increase one while decreasing the other.

### Describe alternatives you have considered.

Adding `"melee_bash"` as a monster object.   It was significantly less work. #57909

**Comment by dseguin** [comment]
Here's a possible solution

<details>
<summary>Patch:</summary>

```diff
diff --git a/src/monstergenerator.cpp b/src/monstergenerator.cpp
index ff15bc0b86..66152545ff 100644
--- a/src/monstergenerator.cpp
+++ b/src/monstergenerator.cpp
@@ -902,7 +902,45 @@ void mtype::load( const JsonObject &jo, const std::string &src )
     if( jo.has_array( "melee_damage" ) ) {
         melee_damage = load_damage_instance( jo.get_array( "melee_damage" ) );
     } else if( jo.has_object( "melee_damage" ) ) {
-        melee_damage = load_damage_instance( jo );
+        melee_damage = load_damage_instance( jo.get_object( "melee_damage" ) );
+    } else if( jo.has_object( "relative" ) ) {
+        cata::optional<damage_instance> tmp_dmg;
+        JsonObject rel = jo.get_object( "relative" );
+        rel.allow_omitted_members();
+        if( rel.has_array( "melee_damage" ) ) {
+            tmp_dmg = load_damage_instance( rel.get_array( "melee_damage" ) );
+        } else if( rel.has_object( "melee_damage" ) ) {
+            tmp_dmg = load_damage_instance( rel.get_object( "melee_damage" ) );
+        } else if( rel.has_int( "melee_damage" ) ) {
+            const int rel_amt = rel.get_int( "melee_damage" );
+            for( damage_unit &du : melee_damage ) {
+                du.amount += rel_amt;
+            }
+        }
+        if( !!tmp_dmg ) {
+            melee_damage.add( tmp_dmg.value() );
+        }
+    } else if( jo.has_object( "proportional" ) ) {
+        cata::optional<damage_instance> tmp_dmg;
+        JsonObject prop = jo.get_object( "proportional" );
+        prop.allow_omitted_members();
+        if( prop.has_array( "melee_damage" ) ) {
+            tmp_dmg = load_damage_instance( prop.get_array( "melee_damage" ) );
+        } else if( prop.has_object( "melee_damage" ) ) {
+            tmp_dmg = load_damage_instance( prop.get_object( "melee_damage" ) );
+        } else if( prop.has_float( "melee_damage" ) ) {
+            melee_damage.mult_damage( prop.get_float( "melee_damage" ), true );
+        }
+        if( !!tmp_dmg ) {
+            for( const damage_unit &du : tmp_dmg.value() ) {
+                auto iter = std::find_if( melee_damage.begin(), melee_damage.end(), [&du]( const damage_unit &mdu ) {
+                    return mdu.type == du.type;
+                } );
+                if( iter != melee_damage.end() ) {
+                    iter->amount *= du.amount;
+                }
+            }
+        }
     }
 
     if( jo.has_array( "scents_tracked" ) ) {
```

</details>

<details>
<summary>And the usage would look like this:</summary>

```JSON
{
  "//": "Proportional usage",
  "proportional": { "melee_damage": [ { "damage_type": "cut", "amount": 0.8 } ] },
  "//": "or",
  "proportional": { "melee_damage": 0.8 },

  "//": "Relative usage",
  "relative": { "melee_damage": [ { "damage_type": "cut", "amount": 2 } ] }
  "//": "or",
  "relative": { "melee_damage": 2 },
}
```

</details>

**Comment by dseguin** [comment]
Fixed by #57957
